crates/amaru/src/bin/ledger/cmd/mithril.rs

```diff
@@ -0,0 +1,392 @@
+        },
+        NetworkName::Preprod => AggregatorDetails {
+            endpoint: "https://aggregator.release-preprod.api.mithril.network/aggregator",
+            verification_key: "5b3132372c37332c3132342c3136312c362c3133372c3133312c3231332c3230372c3131372c3139382c38352c3137362c3139392c3136322c3234312c36382c3132332c3131392c3134352c31332c3233322c3234332c34392c3232392c322c3234392c3230352c3230352c33392c3233352c34345d",
+            ancillary_verification_key: "5b3138392c3139322c3231362c3135302c3131342c3231362c3233372c3231302c34352c31382c32312c3139362c3230382c3234362c3134362c322c3235322c3234332c3235312c3139372c32382c3135372c3230342c3134352c33302c31342c3232382c3136382c3132392c38332c3133362c33365d",
+        },
+        NetworkName::Preview => AggregatorDetails {
+            endpoint: "https://aggregator.testing-preview.api.mithril.network/aggregator",
+            verification_key: "5b3132372c37332c3132342c3136312c362c3133372c3133312c3231332c3230372c3131372c3139382c38352c3137362c3139392c3136322c3234312c36382c3132332c3131392c3134352c31332c3233322c3234332c34392c3232392c322c3234392c3230352c3230352c33392c3233352c34345d",
+            ancillary_verification_key: "5b3138392c3139322c3231362c3135302c3131342c3231362c3233372c3231302c34352c31382c32312c3139362c3230382c3234362c3134362c322c3235322c3234332c3235312c3139372c32382c3135372c3230342c3134352c33302c31342c3232382c3136382c3132392c38332c3133362c33365d",
+        },
+        NetworkName::Testnet(_) => panic!("Mithril not supported on testnets"),
+    }
+}
+
+async fn download_from_mithril(
+    network: NetworkName,
+    target_dir: PathBuf,
+    from_chunk: u64,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let progress_bar = indicatif::MultiProgress::new();
+    let AggregatorDetails {
+        endpoint,
+        verification_key,
+        ancillary_verification_key,
+    } = aggregator_details(network);
+    let client = ClientBuilder::aggregator(endpoint, verification_key)
+        .set_ancillary_verification_key(ancillary_verification_key.to_string())
+        .add_feedback_receiver(Arc::new(IndicatifFeedbackReceiver::new(&progress_bar)))
+        .build()?;
+    let database_client = client.cardano_database_v2();
+    let snapshots = database_client.list().await?;
+    let snapshot_list_item = snapshots.first().ok_or("no snapshot found")?;
+
+    info!(
+        "Downloading and verifying mithril snapshot {}",
+        snapshot_list_item.hash
+    );
+
+    let snapshot = database_client
+        .get(&snapshot_list_item.hash)
+        .await?
+        .ok_or(format!("snapshot not found {}", snapshot_list_item.hash))?;
+    let certificate = client
+        .certificate()
+        .verify_chain(&snapshot.certificate_hash)
+        .await?;
+
+    let immutable_file_range = ImmutableFileRange::From(from_chunk);
+    let download_unpack_options = DownloadUnpackOptions {
+        allow_override: true,
+        include_ancillary: false,
```

Comment:
So you don't need the ledger state for the bootstrap of Amaru? 

- If you need it, you would need to include ancillaries:
```suggestion
        include_ancillary: true,
```

- Otherwise, you can probably get rid of the `ancillary_verification_key` in the `AggregatorDetails` which is needed only if `include_ancillary` is set to `true`.